### PR TITLE
Override templates by php version

### PR DESCRIPTION
**Pull request by elythyr**
Fixes https://github.com/phpactor/phpactor/issues/856

https://github.com/phpactor/phpactor/pull/857 need this PR to be merged in order for the CI to pass.

### CODE REVIEW

**Comment by dantleech** [comment]
> It needs phpactor/phpactor#857 to be merged in order to have access to the PHP version override

I guess this isn't true? i.e. Phpactor depends on this package, this package doesn't depend on Phpactor.

File: lib/Domain/Helper/TemplatePathsResolver.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Phpactor\CodeBuilder\Domain\Helper;
+
+use FilesystemIterator;
+
+class TemplatePathsResolver
+{
+    /**
+     * @var string In the form of "major.minor.release[extra]"
```
Comment (dantleech) [inline lib/Domain/Helper/TemplatePathsResolver.php]: Is this a valid docblock? Otherwise I'd be happy to ommit the `@var`

**Comment by elythyr** [comment]
> > It needs [phpactor/phpactor#857](https://github.com/phpactor/phpactor/pull/857) to be merged in order to have access to the PHP version override
> 
> I guess this isn't true? i.e. Phpactor depends on this package, this package doesn't depend on Phpactor.

Yes, it's the other way around, I updated it and the message in the PR on phpactor.

File: lib/Domain/Helper/TemplatePathsResolver.php (hunk 1 above)
Comment (elythyr) [reply lib/Domain/Helper/TemplatePathsResolver.php]: I think so: https://docs.phpdoc.org/references/phpdoc/tags/var.html

**Comment by dantleech** [comment]
I made a PR on this PR which renames the resolver and adds an integration test: https://github.com/elythyr/code-builder/pull/1

File: lib/Domain/Helper/TemplatePathsResolver.php (hunk 1 above)
Comment (dantleech) [reply lib/Domain/Helper/TemplatePathsResolver.php]: > The @var tag MUST contain the name of the element it documents. An exception to this is when property declarations only refer to a single property. In this case the name of the property MAY be omitted.

Only can be omitted for properties at the moment.

**Comment by elythyr** [comment]
Nice !

**Comment by dantleech** [comment]
Looks like my change caused the CS to fail..

**Comment by elythyr** [comment]
Yeah I really need to had some git hook before pushing...

**Comment by dantleech** [comment]
:tada:

**Pull request merged**